server/blob/blob-s3/src/main/java/org/apache/james/blob/objectstorage/aws/S3DumbBlobStore.java

```diff
@@ -99,12 +103,21 @@
             .region(region.asAws())
             .serviceConfiguration(pathStyleAccess)
             .build();
+
+        clientPool = PoolBuilder.from(Mono.fromCallable(clientCreator))
+            .acquisitionScheduler(Schedulers.elastic())
+            .destroyHandler(client -> Mono.fromRunnable(client::close))
+            .maxPendingAcquireUnbounded()
+            .sizeBetween(4, 64)
```

Comment:
If possible let's have an unbounded pool, and make the size configurable with ridiculously high defaults otherwize.

Reply:
Yes I saw your comments on the related PR

There was a comment regarding the warmup, to not do it in constructor but Startable pattern fashion too